scripts/start.js

```diff
@@ -14,13 +14,22 @@ const { hideBin } = require('yargs/helpers');
 const message = require('./utils/message');
 
 message.title('start dev');
-
 const argv = yargs(hideBin(process.argv)).argv;
+
+const preStart = path.join(__dirname, '..', 'types', argv.type, 'scripts', 'preStart.js');
```

Comment:
@allyoucanmap could you explain briefly the need for preCompile and preStart scripts?

Reply:
@mbarto I was thinking that they could replace the clean.js script and be more generic for other pourposes. In this case the name of the extension is controlled by a single variable inside the package.json configuration (mapstore.extensionName). This name is injected everywhere from the webpack so we don't need to manually replace in all the places. The only file I was not able to override dynamically was the index.json of an extension for this reason just before the start or compile it checks if the index.json exists and it's name match the one in package.json if not it will create the index.json with the right info ([preStart.js](https://github.com/geosolutions-it/mapstore-project/pull/13/files#diff-2ce5ad7fab1aeb9dffc49f4e962e023eae72cbfad5712eaf0ddbba6869ee0efcR9-R25) of extension).
Maybe we could find a better name/implementation for those...

Reply:
got it, the name looks good to me, thx